pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -591,7 +591,7 @@ func takeByTopologyNUMADistributed(topo *topology.CPUTopology, availableCPUs cpu
 		acc.iterateCombinations(numas, k, func(combo []int) LoopControl {
 			// If we've already found a combo with a balance of 0 in a
 			// different iteration, then don't bother checking any others.
-			if bestBalance == 0 {
+			if bestBalance == 0 && bestBalanceInOneSocket {
```

Comment:
this is set to true only if alignBySocket is true. Do we have a bug? or are we "just" doing more iterations than before?